File: EduTrailblaze/EduTrailblaze.Entities/Lecture.cs
using EduTrailblaze.API.Domain;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EduTrailblaze.Entities
{
    public class Lecture : EntityAuditBase<int>
    {
        [Required, ForeignKey("Section")]
        public int SectionId { get; set; }

        [Required, StringLength(50)]
        public string Title { get; set; }

        [Required, StringLength(int.MaxValue)]
        public string Content { get; set; }

        [Required, StringLength(int.MaxValue)]
        public string Description { get; set; }

        [Required]
        public int Duration { get; set; }

        public bool IsDeleted { get; set; } = false;



        // Navigation properties
        public virtual Section Section { get; set; }
        public virtual ICollection<Quiz> Quizzes { get; set; }
        public virtual ICollection<Video> Videos { get; set; }
        public virtual ICollection<UserProgress> UserProgresses { get; set; }
    }
}

File: EduTrailBlaze-FE/src/types/lecture.ts
interface SectionLecture {
  sectionId: number
  lectures: ILecture[]
}

interface ILecture {
  sectionId: number | null
  title: string
  content: string
  duration: string
  description: string
  isDeleted: boolean
  createdAt: string
  updatedAt: string
  id: number
  type: LectureType
}
enum LectureType {
  Reading = 'Reading',
  Video = 'Video',
  Quiz = 'Quiz'
}
interface ISection {
  courseId: number
  title: string
  description: string
  numberOfLectures: number
  duration: string
  id: number
}
